### PR TITLE
Serverless offline 6

### LINKED ISSUE #116: serverless-offline-sqs incompatible with new versions of serverless-offline
When installing alongside newer versions of serverless-offline this line:

`const LambdaContext = require('serverless-offline/src/LambdaContext');`

fails as LambdaContext doesn't exist on that path

### PR DESCRIPTION
**Pull request by esetnik**
Resolves #116, #122 

Note that to achieve this there's a pretty big hack. I need to access the private class property #lambda. See https://github.com/CoorpAcademy/serverless-plugins/pull/125/files#diff-32e9d7d10726a2f50f1aea57d090602cR196

### CODE REVIEW

**Comment by esetnik** [comment]
Note for maintainers. I can’t get the python test to pass. Can someone who knows about running python on serverless take a look?

File: packages/serverless-offline-sqs/src/index.js [hunk 1]
```diff
@@ -56,6 +53,14 @@ class ServerlessOfflineSQS {
       'before:offline:start:end': this.offlineStartEnd.bind(this)
     };
 
+    this.serverlessOfflinePlugin = this.serverless.pluginManager.plugins.find(
```
Comment (bakasmarius) [inline packages/serverless-offline-sqs/src/index.js:56]: For me, this doesn't find anything even though I do have `serverless-offline-sqs` plugin installed... Are you sure this is the best way to find it?

Comment (esetnik) [reply packages/serverless-offline-sqs/src/index.js:56]: I couldn't find any better way to hook into the serverless-offline plugin. The plugin order matters. Make sure that `- serverless-offline` appears before `- serverless-offline-sqs` in your `serverless.yml`

Comment (bakasmarius) [reply packages/serverless-offline-sqs/src/index.js:56]: Yup, that was it - after changing the order I can approve that it works on my side. (tested with serverless-offline versions 6.1.0 and 6.1.4)

**Comment by esetnik** [comment]
@AdrieanKhisbe can you take a look please?

**Comment by clar-cmp** [comment]
Any plans to look at this soon?

**Comment by johanpedroo** [comment]
waiting for this :(

**Comment by YOU54F** [comment]
I changed the lines https://github.com/CoorpAcademy/serverless-plugins/blob/fc19ef6d8a10bde712adf7b95f0ede3fa673a0d8/packages/serverless-offline-sqs/src/index.js#L27 in the source under node_modules tto

```
const functionHelper = require('serverless-offline/dist/lambda/LambdaFunction');
const LambdaContext = require('serverless-offline/dist/lambda/LambdaContext');
```

and it looks to have created my queue against serverless-offline `v6.3.0` 

I published a version on npm as `npm install serverless-offline-sqs-elastic` as `4.0.0` 

Will drop a pull request after some more testing.

**Comment by YOU54F** [comment]
Thanks for this @esetnik, this is working for me against serverless-offline `v6.3.0` running against a node 12 service running locally

I published a version on npm as `npm i serverless-offline-sqs-elastic` whilst we wait for this to get merged into a main branch.

https://www.npmjs.com/package/serverless-offline-sqs-elastic

**Comment by christophersbarrett** [comment]
@YOU54F your npm package points back to this repo though?

**Comment by godu** [comment]
Hi, i'm working on full rewrite to support serverless v6.
Take a look of https://github.com/CoorpAcademy/serverless-plugins/pull/129

**Pull request closed without merging**